modules/javafx.graphics/src/main/java/com/sun/javafx/iio/ImageStorage.java

```diff
@@ -433,9 +483,11 @@ private static synchronized int getMaxSignatureLength() {
 //        return loader;
 //    }
 
-    private static ImageLoader getLoaderBySignature(InputStream stream, ImageLoadListener listener) throws IOException {
+    private ImageLoader getLoaderBySignature(InputStream stream, ImageLoadListener listener) throws IOException {
         byte[] header = new byte[getMaxSignatureLength()];
-        ImageTools.readFully(stream, header);
+        if (stream.read(header) <= 0) {
+            return null;
+        }
```

Comment:
What was the reason for this change? The former would work even if the stream returned less data that the size of the header in a single read, whereas the latter would fail.

Reply:
Well, almost. `ImageTools.readFully` throws `EOFException` if the input stream is empty. That doesn't make a lot of sense in the context of `getLoaderBySignature`, since the method should return `null` if the input stream doesn't contain enough data to detect a signature (and it doesn't seem useful to me to make a distinction between "not enough" bytes and zero bytes). I've opted to retain the use of `ImageTools.readFully`, but catch the `EOFException` and return `null` instead.